.circleci/config.yml

```diff
@@ -124,6 +124,25 @@ jobs:
       - run: |
           kurtosis run ./simple-api/kurtosis-package/
 
+  build_deploy_simple_api:
+    executor: ubuntu_vm
+    steps:
+      - checkout
+      # If this sub-repo hasn't changed, don't run the tests
+      - run: |
+          if git --no-pager diff --exit-code origin/main...HEAD -- ':.circleci' ':simple-api'; then
+            circleci-agent step halt
+          fi
+      - run: echo "${DOCKER_PASSWORD}" | docker login -u ${DOCKER_USERNAME} --password-stdin
+      - run: simple-api/scripts/build.sh
+      - run: |
+          set -euo pipefail
+          version_to_publish="latest"
```

Comment:
Interesting, we don't care about versioning here? Just to make sure we're on the same page, this will basically override the docker image everytime, trashing the previous version

Reply:
and if this happens with every push to any PR then its quite risky as someone might publish something that doesn't work and then the user (interview candidate in this case is left with a broken thing)

Reply:
I had actually chatted with @h4ck3rk3y about this a while back and at the time we felt it was overkill to have versioning for this image based on what we are doing. Personally I still think it is, but I'm willing to be convinced otherwise if you both feel strongly. This image is an example that we will likely use in our FE candidate interviewing.

If you break the image the user will be stuck with a broken image. That seems like a general truth though (i.e. applies everywhere). If we use this with candidates I doubt they would just use an earlier version. We should make sure that the images we are pushing work. But that's just my $0.02. Willing to have my mind changed if you feel differently :)

Reply:
We don't explicitly need versioning but we shouldn't be publishing every commit that gets made to any PR on this project as that commit might be broken & the image would get in a bad state.

I'd expect a merge to main to get published and you can tag it as latest & if you want you can also add the latest git commit tag using (git rev-parse --short HEAD)

Reply:
Yes, agreed with Gyani, if we're only publishing on the main branch I'm fine keeping the tag to `latest` for every publish

Reply:
Oh I might add the git commit tag as well. Like that idea. thank you both!